src/DashboardService.Data/PriorityRepository.cs

```diff
@@ -18,28 +21,83 @@ public PriorityRepository(IDataProvider provider)
     _provider = provider;
   }
   
-  public Task<Guid?> CreateAsync(DbPriority dbDepartment)
+  public async Task<Guid?> CreateAsync(DbPriority dbPriority)
   {
-    throw new NotImplementedException();
+    _provider.Priorities.Add(dbPriority);
+    await _provider.SaveAsync();
+
+    return dbPriority.Id;
+  }
+
+  public async Task<(List<DbPriority> dbPriorities, int totalCount)> GetAllAsync(GetPrioritiesFilter filter, CancellationToken ct)
+  {
+    IQueryable<DbPriority> dbPriorities = _provider.Priorities.AsNoTracking();
+    
+    if (filter.IsAscendingSort.HasValue)
+    {
+      dbPriorities = filter.IsAscendingSort.Value
+        ? dbPriorities.OrderBy(d => d.Name)
+        : dbPriorities.OrderByDescending(d => d.Name);
+    }
+    
+    if (!string.IsNullOrWhiteSpace(filter.NameIncludeSubstring))
+    {
+      dbPriorities = dbPriorities.Where(d => d.Name.Contains(filter.NameIncludeSubstring));
+    }
+    
+    return (
+      await dbPriorities
+        .Skip(filter.SkipCount)
+        .Take(filter.TakeCount)
+        .ToListAsync(ct),
+      await dbPriorities.CountAsync(ct));
+  }
+
+  public async Task<DbPriority> GetAsync(Guid id, CancellationToken ct = default)
+  {
+    return await _provider.Priorities.AsNoTracking().SingleOrDefaultAsync(p=>p.Id == id, ct);
   }
 
-  public Task<List<DbPriority>> GetAllAsync(GetPrioritiesFilter filter)
+  public async Task<bool> EditAsync(Guid id, JsonPatchDocument<DbPriority> request)
   {
-    throw new NotImplementedException();
+    DbPriority dbPriority = await GetAsync(id);
+
+    if (dbPriority is null)
+    {
+      return false;
+    }
+    
+    request.ApplyTo(dbPriority);
```

Comment:
Do you test this method by postman/swagger?

Reply:
yes